src/Bicep.LangServer/Completions/PublicRegistryModuleMetadataProvider.cs

```diff
@@ -16,69 +17,124 @@ namespace Bicep.LanguageServer.Providers
     public record ModuleMetadata(string moduleName, List<string> tags);
 
     /// <summary>
-    /// Provider to get modules metadata from this endpoint - https://live-data.bicep.azure.com/modulesMetadata
-    /// The above endpoint helps fetch module names and versions of modules available in this github repository - https://github.com/Azure/bicep-registry-modules
+    /// Provider to get modules metadata that we store at a public endpoint.
+    /// This endpoint caches module names and versions of modules available in this github repository - https://github.com/Azure/bicep-registry-modules
     /// </summary>
     public class PublicRegistryModuleMetadataProvider : IPublicRegistryModuleMetadataProvider
     {
-        private const string LiveDataEndpoint = "https://live-data.bicep.azure.com/modulesMetadata";
-        private List<ModuleMetadata> moduleMetadataCache = new List<ModuleMetadata>();
+        private const string LiveDataEndpoint = "https://aka.ms/BicepModulesMetadata";
+        private readonly TimeSpan cacheValidFor = TimeSpan.FromHours(1);
 
-        public async Task Initialize()
+        private ModuleMetadata[] cachedModules = new ModuleMetadata[] { };
+        private bool isQueryingLiveData = false;
+        private object queryingLiveSyncObject = new();
+        private DateTime? lastSuccessfulQuery;
+
+        public PublicRegistryModuleMetadataProvider(bool initializeCache = false)
         {
-            try
+            if (initializeCache)
             {
-                HttpClient httpClient = new HttpClient();
-                var moduleMetadata = await httpClient.GetStringAsync(LiveDataEndpoint);
-                var metadata = JsonConvert.DeserializeObject<List<ModuleMetadata>>(moduleMetadata);
+                this.UpdateCacheAsync(true);
+            }
+        }
 
-                if (metadata is not null)
+        private void UpdateCacheAsync(bool initialDelay = false)
```

Comment:
What's the reason for the delay?

Reply:
TBH it's mostly because Trace doesn't seem to work early on.

Reply:
Seems I lost the comment I had had there.

Reply:
Ah, makes sense. Probably because we wait for initialization before configuring logging:

https://github.com/Azure/bicep/blob/03d87187828709d4d966dc8b06f97260f8e59e3c/src/Bicep.LangServer/Server.cs#L86-L89